test/extended/router/idle.go

```diff
@@ -145,3 +147,56 @@ func checkIdleAnnotationValues(annotations map[string]string) {
 		},
 	}))
 }
+
+// waitForRunningPods waits for podCount pods matching podSelector are
+// in the running state. It retries the request every second and will
+// return an error if the conditions are not met after the specified
+// timeout.
+func waitForRunningPods(oc *exutil.CLI, podCount int, podLabels labels.Selector, timeout time.Duration) error {
+	ns := oc.KubeFramework().Namespace.Name
+
+	if err := wait.PollImmediate(time.Second, timeout, func() (bool, error) {
+		podList, err := oc.AdminKubeClient().CoreV1().Pods(ns).List(context.Background(), metav1.ListOptions{LabelSelector: podLabels.String()})
+		if err != nil {
+			e2e.Logf("Error listing pods: %v", err)
+			return false, nil
+		}
+		return len(podList.Items) == podCount, nil
+	}); err != nil {
+		return fmt.Errorf("failed to list pods: %v", err)
+	}
+
+	e2e.Logf("Waiting for %d pods in namespace %s", podCount, ns)
+	c := oc.AdminKubeClient()
```

Comment:
I'm curious why you don't make this call to `AdminKubeClient` just once and pass it to the anonymous function in line 158, but I know it passed the first code review.

Reply:
As this is an automatic cherry-pick I think anything that needs doing here would be better off in master.
And... I can only think I did it this way to begin with for readability.

Reply:
> I'm curious why you don't make this call to `AdminKubeClient` just once and pass it to the anonymous function in line 158, but I know it passed the first code review.

Looking at this again; it's only used below so wouldn't need passing to "line 158". I think I did it this way for readability, though that's a bit tenuous as it is only used on line 171.